tests/properties_test.py

```diff
@@ -0,0 +1,244 @@
+            fromi_ref = state_to_state[f"from_{i}"]["transition_dipole_moments"]
+
+            state2state = State2States(state, initial=i)
+            for ii, j in enumerate(range(i + 1, state.size)):
+                assert state.excitation_energy[j] == refevals[j]
+                assert_allclose_signfix(state2state.transition_dipole_moment[ii],
+                                        fromi_ref[ii], atol=1e-4)
+
+    @pytest.mark.parametrize("case", ["gen", "cvs"])
+    def test_magnetic_transition_dipole_moments_z_component(self, method: str,
+                                                            case: str):
+        backend = ""
+        xyz = """
+            C 0 0 0
+            O 0 0 2.7023
+        """
+        basis = "sto-3g"
+        scfres = run_hf(backend, xyz, basis)
+
+        if "cvs" in case:
+            if "cvs" not in method:
+                method = f"cvs-{method}"
+            state = run_adc(scfres, method=method, n_singlets=5, core_orbitals=2)
+        else:
+            state = run_adc(scfres, method=method, n_singlets=10)
+        tdms = state.transition_magnetic_dipole_moment
+
+        # For linear molecules lying on the z-axis, the z-component must be zero
+        for tdm in tdms:
+            assert tdm[2] < 1e-10
+
+    # Only adcc reference data available.
+    @pytest.mark.parametrize("system,case,kind", cases)
+    def test_magnetic_transition_dipole_moments(self, system: str, case: str,
+                                                kind: str, method: str):
+        refdata = testdata_cache._load_data(
+            system=system, method=method, case=case, source="adcc"
+        )[kind]
+        state = testdata_cache._make_mock_adc_state(
+            system=system, method=method, case=case, kind=kind, source="adcc"
+        )
+
+        res_dms = state.transition_magnetic_dipole_moment
+        n_ref = len(state.excitation_vector)
+        assert_allclose(
+            res_dms, refdata["transition_magnetic_dipole_moments"][:n_ref],
+            atol=1e-4
+        )
+
+    # Only adcc reference data available.
+    @pytest.mark.parametrize("system,case,kind", cases)
+    def test_transition_dipole_moments_velocity(self, system: str, case: str,
+                                                kind: str, method: str):
+        refdata = testdata_cache._load_data(
+            system=system, method=method, case=case, source="adcc"
+        )[kind]
+        state = testdata_cache._make_mock_adc_state(
+            system=system, method=method, case=case, kind=kind, source="adcc"
+        )
+
+        res_dms = state.transition_dipole_moment_velocity
+        n_ref = len(state.excitation_vector)
+        assert_allclose(
+            res_dms, refdata["transition_dipole_moments_velocity"][:n_ref],
+            atol=1e-4
+        )
+
+    # Only adcc reference data available.
+    @pytest.mark.parametrize("system,case,kind", cases)
+    def test_rotatory_strengths(self, system: str, case: str, kind: str,
```

Comment:
This should be zero for everything that is not chiral. Should we also test it for something chiral like methyloxirane?

Reply:
You are the expert 😄
Mhhh... In this test, we verify that the rotatory strength is correctly calculated from the electronic and magnetic transition dipole moments. In principle testing for a zero result is also a verification. But testing for a nonzero result is probably better.
However, at the moment we don't have any MP/ADC reference data for methyloxirane. So we would have to generate, store and maintain the test data exclusively for this test.

Reply:
Okay, then let's leave it for now.